### DISCUSSION TITLE
EP2336

### DISCUSSION THREAD

**Discussion by benceje** [TillFleisch/ESPHome-Philips-Smart-Coffee#79] [Support for EPxxxx]
Hello Till,
I tested it with the EP2336 device, which is very similar to the EP2235 device in terms of functions and button layout.
All selections work as they should, quantity setting start, on and off. The only mistake is that for hot water and cappuccino, the water does not come out at the place of the milk holder, but directly into the water tray through the valve behind the removable middle part, as if cleaning it. If the module is inside, it does this even when started from the control panel. I remove the panel, everything works fine. In your opinion, could there be a different command for this type for the two mentioned functions?
I saw in topic #54 that there was the same problem, but I didn't actually read a good solution. Or did I miss something?

**Comment by TillFleisch** [comment]
> The only mistake is that for hot water and cappuccino, the water does not come out at the place of the milk holder, but directly into the water tray through the valve behind the removable middle part, as if cleaning it. 

As far as my understanding goes, there are some devices which already have 'smart functionality' (through their App/Cloud/whatever). I have never been hands-on with one of those machines, but according to the specification (or their advertising) the machine already offers a "clean the machine without filling the cup" feature. This is arguably even better than the solution that this repo offers, which just skips the cleaning step.

It would make sense for the machine to have different "commands" for this feature, under the assumption that the "smart parts" are on the display side. If this is not the case, there is no need for the display to know whether the machine has started with or without (internal) cleaning. Thus (ab-)using this functionality through the ESP may not be possible.

It seems to be the case that these messages contain the model NR in bytes 4-6 (zero-indexed). 
EP2220
`0xD5, 0x55, 0x00, 0x01, 0x02, 0x00, 0x02, 0x01, 0x00, 0x00, 0x1D, 0x3B`

EP32xx
`0xD5, 0x55, 0x00, 0x01, 0x03, 0x00, 0x0E, 0x01, 0x00, 0x00, 0x39, 0x39`

I suspect that these are different on your machine (EP2336), somewhere along the lines of `0x02, 0x00, 0x??`.
This leads me to believe that each model is running different code/a different configuration. Therefore, separate "commands" for the missing features are not unfeasible.

Retrieving these commands may be possible using the UART debug functionality and it may require (depending on how things are built) the use of the App, to hopefully retrieve commands which are not available via the display panel directly.

**Comment by benceje** [comment]
Thank you very much for your reply. I expected something similar; I knew we
were going to get to UART debugging, that's what I was dreading. I've never
used it, although I know arduino... Do you need a wemos d1 mini for this?
(it's at home) I think the ESP12F PCB doesn't know this, or at least not
with the flash content used for HA control.

**Comment by TillFleisch** [comment]
Assuming you already have a modified machine, all you'd need to do is to change your YAML configuration. See [UART Debug documentation](https://esphome.io/components/uart.html#debugging). Usually, only observing "RX" on the display UART can provide enough information. But then again I don't know where the existing "smart parts" are so...

Depending on how 'complicated' the extended protocol is, adding support for the features may require some work. Without access to one of the "smart" models I can unfortunately not promise help/support

**Comment by benceje** [comment]
Yes, there is the modification in the coffee maker, the ESP12F chipped PCB is included with connectors, no cut cables.
Do you think this will be enough to read the codes in the ESP yaml configuration? Or do you need something else?
I think I can leave the head part, I didn't even copy it here, and for the transmission to work, the PCB needs the power_pin, delay, etc.

captive_portal:

uart:
  debug:
    direction: BOTH
    dummy_receiver: false
    after:
      delimiter: "\n"
    sequence:
      - lambda: UARTDebug::log_string(direction, bytes);
  - tx_pin: GPIO1
    rx_pin: GPIO3
    baud_rate: 115200
    id: uart_mainboard
  - tx_pin: GPIO15
    rx_pin: GPIO13
    baud_rate: 115200
    id: uart_display

philips_coffee_machine:
  display_uart: uart_display
  mainboard_uart: uart_mainboard
  power_pin: GPIO12
  power_trip_delay: 750ms
  id: coffee_machine
  model: EP_2235
  power_message_repetitions: 25
  language: en-US

**Comment by TillFleisch** [comment]
You can use your default configuration. All you need to add is the following section (under the UART section):
```
  - tx_pin: GPIO15
    rx_pin: GPIO13
    baud_rate: 115200
    id: uart_display
    debug:
      direction: BOTH
      sequence:
        - lambda: UARTDebug::log_hex(direction, bytes, ' ');
```

You could also use the method laid out in https://github.com/TillFleisch/ESPHome-Philips-Smart-Coffee/issues/7#issuecomment-1941624536, which already filters different commands (but it can also filter out some valid ones).

**Comment by benceje** [comment]
Thanks, I'll check it out tonight, but I have a question. Will I see the
data in ESP LOG under HA, or do I need a separate program interface?
Can I ask for more help?

**Comment by TillFleisch** [comment]
You can use the ESPHome Dashboard (the HA-Addon) to view the logs, as with any other ESPHome device.
If you are feeling fancy, you can also use the ESPHome command line interface to view logs.

I would recommend viewing logs over-the-air.

**Comment by benceje** [comment]
Sorry, one more question. I read through what you linked (#7 (comment)). It is not clear to me: Now should I leave all lines in the yaml file as I put together for the HA control with all switches and sensors and only write what you wrote under the display uart line?
Or just what you wrote in the email? For example, will the device turn on at the press of a button if the definition of 
 POWER_MESSAGE_REPETITIONS and the power pin is not included in the yaml config?
Or should I just leave it at this:
philips_coffee_machine:
  display_uart: uart_display
  mainboard_uart: uart_mainboard
  power_pin: GPIO12
  power_trip_delay: 750ms
  id: coffee_machine
  model: EP_2235
  power_message_repetitions: 25
  language: en-USips_coffee_machine:

Because if the module is inside, it also works incorrectly from the physical button...

**Comment by TillFleisch** [comment]
You leave the `philips_coffee_machine` section as is. Only the uart section (and globals in case of the related commend) need to be changed.

**Comment by benceje** [comment]
I did it... This is the log file, I don't really understand what I should see. It is certain that in the case of cappuccino and tea, the hot water does not come out of the milk container.
[logs_philis_logs (2).txt](https://github.com/user-attachments/files/16765527/logs_philis_logs.2.txt)

**Comment by nono721987** [comment]
Hello, I also have the 2336 machine. For me, it's already failing because the display doesn't turn on. When I press power on the ESP, the machine starts, but the display remains dark. Would you mind sending me your wiring and which PCB you used? Thanks.

**Reply by benceje** [reply]
Hello! I used the Custom PCB (which is also mentioned in the description), but I haven't dealt with it since then because there is no solution to the problem. The 2236 uses different commands...

**Reply by nono721987** [reply]
Do you mean this PCB? https://oshwlab.com/tradstaat/coffeepcb I have that, but as I said, the display doesn't turn on. Even when I try to start the display with a broken cable, it doesn't work. You used the code 2335 for testing, right?

**Reply by benceje** [reply]
I didn't use the D1 Mini version, but the newer one, this one:
https://oshwlab.com/tillf/philips-smart-coffee-pcb
Yes, I used the code 2335 as the base.

**Reply by nono721987** [reply]
Ok Thank you.
